src/Npgsql/NpgsqlConnector.cs

```diff
@@ -1326,22 +1332,27 @@ internal void Break()
                     // Note that the connection's full state is usually calculated from the connector's, but in
                     // states closed/broken the connector is null. We therefore need a way to distinguish between
                     // Closed and Broken on the connection.
-                     conn.Close(true);
+                     conn.Close(wasBroken: true, async: false);
                 }
             }
         }
 
         /// <summary>
         /// Closes the socket and cleans up client-side resources associated with this connector.
         /// </summary>
+        /// <remarks>
+        /// This method doesn't actually perform any meaningful I/O, and therefore is sync-only.
+        /// </remarks>
         void Cleanup()
         {
             Debug.Assert(Monitor.IsEntered(this));
 
             Log.Trace("Cleaning up connector", Id);
             try
             {
-                _stream?.Dispose();
+                var stream = _stream;
+                if (stream != null)
+                    _stream.Dispose();
```

Comment:
And this. Any reason to expand the expression when the compiler will do it for you?

Reply:
You're right, this is unnecessary (it made sense at some point). Will remove.